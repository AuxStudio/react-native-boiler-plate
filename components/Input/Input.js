import React from 'react';
import PropTypes from 'prop-types';
import { TextInput, Text, ViewPropTypes } from 'react-native';
import { TextField } from 'react-native-material-textfield';

import styleConstants from '../../styleConstants';

const propTypes = {
  value: PropTypes.string,
  handleChangeText: PropTypes.func,
  label: PropTypes.string.isRequired,
  title: PropTypes.string,
  error: PropTypes.string,
  characterRestriction: PropTypes.number,
  disabled: PropTypes.bool,
  baseColor: PropTypes.string,
  tintColor: PropTypes.string,
  labelTextStyle: Text.propTypes.style,
  titleTextStyle: Text.propTypes.style, // also errorTextStyle
  style: TextInput.propTypes.style,
  containerStyle: ViewPropTypes.style,
  inputContainerStyle: ViewPropTypes.style,
  renderAccessory: PropTypes.func,

  autoFocus: PropTypes.bool,
  keyboardType: PropTypes.string, // default, numeric, email-address, phone-pad
  multiline: PropTypes.bool,
  returnKeyType: PropTypes.string, // done, go, next, search, send
  secureTextEntry: PropTypes.bool,
  handleBlur: PropTypes.func,
  handleFocus: PropTypes.func,
  handleSubmit: PropTypes.func,
};

const defaultProps = {
  tintColor: styleConstants.colors.primary,
};

const Input = ({
  value,
  handleChangeText,
  label,
  title,
  error,
  characterRestriction,
  disabled,
  baseColor,
  tintColor,
  labelTextStyle,
  titleTextStyle,
  style,
  containerStyle,
  inputContainerStyle,
  renderAccessory,

  autoFocus,
  keyboardType,
  multiline,
  returnKeyType,
  secureTextEntry,
  handleBlur,
  handleFocus,
  handleSubmit,
}) => {
  return (
    <TextField
      value={value}
      onChangeText={(text) => handleChangeText(text)}
      label={label}
      title={title}
      error={error}
      characterRestriction={characterRestriction}
      disabled={disabled}
      baseColor={baseColor}
      tintColor={tintColor}
      labelTextStyle={labelTextStyle}
      titleTextStyle={titleTextStyle}
      style={style}
      containerStyle={containerStyle}
      inputContainerStyle={inputContainerStyle}
      renderAccessory={renderAccessory}
      autoFocus={autoFocus}
      keyboardType={keyboardType}
      multiline={multiline}
      returnKeyType={returnKeyType}
      secureTextEntry={secureTextEntry}
      onBlur={handleBlur}
      onFocus={handleFocus}
      onSubmitEditing={handleSubmit}
    />
  );
};

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;
