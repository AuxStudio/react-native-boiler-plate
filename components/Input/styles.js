import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({});

export default styles;
