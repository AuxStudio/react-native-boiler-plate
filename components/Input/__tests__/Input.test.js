import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Input from '..';

describe('Input', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <Input
          value=""
          handleChangeText={jest.fn()}
          label="Test"
          title="Title"
          error="Error"
          characterRestriction={100}
          baseColor="red"
          tintColor="blue"
          labelTextStyle={{ color: 'red' }}
          titleTextStyle={{ color: 'blue' }}
          style={{ backgroundColor: 'green' }}
          containerStyle={{ backgroundColor: 'purple' }}
          inputContainerStyle={{ backgroundColor: 'orange' }}
          keyboardType="numeric"
          multiline
          returnKeyType="search"
          secureTextEntry
          disabled
          renderAccessory={() => <View />}
          handleBlur={jest.fn()}
          handleFocus={jest.fn()}
          handleSubmit={jest.fn()}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<Input label="Test" />)).toMatchSnapshot();
  });
});
