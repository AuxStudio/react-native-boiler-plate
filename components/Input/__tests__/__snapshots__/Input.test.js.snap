// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with all props 1`] = `
<View
  onResponderRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "purple",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "orange",
        "height": 64,
        "overflow": "hidden",
        "paddingBottom": 8,
        "paddingTop": 32,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "borderColor": "red",
            "borderStyle": "dotted",
            "borderWidth": 1,
            "bottom": 0,
            "left": -1.5,
            "right": -1.5,
            "top": -2,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "position": "absolute",
          "top": 36,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(213, 0, 0, 1)",
            "fontSize": 16,
          }
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="sentences"
        disableFullscreenUI={true}
        editable={false}
        error="Error"
        keyboardType="numeric"
        multiline={true}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[MockFunction]}
        renderAccessory={[Function]}
        returnKeyType="search"
        secureTextEntry={true}
        selectionColor="blue"
        style={
          Array [
            Object {
              "flex": 1,
              "margin": 0,
              "padding": 0,
              "top": 2,
            },
            Object {
              "color": "red",
              "fontSize": 16,
              "height": 48,
              "textAlign": "left",
              "top": -1,
            },
            Object {
              "backgroundColor": "green",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "justifyContent": "center",
            "top": 2,
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 24,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "blue",
              "fontSize": 12,
              "opacity": 1,
            }
          }
        >
          Error
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "blue",
              "fontSize": 12,
              "opacity": 0,
            }
          }
        >
          Title
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 4,
          "paddingVertical": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "textAlign": "right",
            },
            Object {
              "color": "blue",
            },
            Object {
              "color": "red",
              "fontSize": 12,
            },
          ]
        }
      >
        0
         / 
        100
      </Text>
    </View>
  </View>
</View>
`;

exports[`Input renders with minimum required props 1`] = `
<View
  onResponderRelease={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(0, 0, 0, 0.38)",
        "borderBottomWidth": 0.5,
        "height": 64,
        "paddingBottom": 8,
        "paddingTop": 32,
      }
    }
  >
    <View
      style={
        Object {
          "position": "absolute",
          "top": 36,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.38)",
            "fontSize": 16,
          }
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="sentences"
        disableFullscreenUI={true}
        editable={true}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        onFocus={[Function]}
        selectionColor="#3D84A8"
        style={
          Array [
            Object {
              "flex": 1,
              "margin": 0,
              "padding": 0,
              "top": 2,
            },
            Object {
              "color": "rgba(0, 0, 0, .87)",
              "fontSize": 16,
              "height": 24,
              "textAlign": "left",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(213, 0, 0)",
              "fontSize": 0,
              "opacity": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": 0,
            "left": 0,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, .38)",
              "fontSize": 12,
              "opacity": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
